Clarify Mac contributing set up instructions

![image](https://user-images.githubusercontent.com/251288/128804711-65a74d0d-f4c5-4e07-b84d-70ace76979d7.png)

I followed the `Environment set up` instructions through `npm ci` and then tried `npm run build` and got `Error: could not find the `cargo` executable.`

<details>

```
npm run build

> typescript-sdk@ build /Users/me/gh/sdk-node
> lerna run --stream build

lerna notice cli v4.0.0
lerna info versioning independent
lerna info Executing command in 6 packages: "npm run build"
@temporalio/core-bridge: > @temporalio/core-bridge@0.2.0 build /Users/me/gh/sdk-node/packages/core-bridge
@temporalio/core-bridge: > npm run build-rust
@temporalio/create: > @temporalio/create@0.3.3 build /Users/me/gh/sdk-node/packages/create-project
@temporalio/create: > tsc --build
@temporalio/core-bridge: > @temporalio/core-bridge@0.2.0 build-rust /Users/me/gh/sdk-node/packages/core-bridge
@temporalio/core-bridge: > TEMPORAL_WORKER_FORCE_BUILD=true node ./scripts/build.js
@temporalio/core-bridge: Force build via TEMPORAL_WORKER_FORCE_BUILD env var
@temporalio/core-bridge: Compiling bridge { target: undefined }
@temporalio/core-bridge: Error: could not find the `cargo` executable.
@temporalio/core-bridge: You can find instructions for installing Rust and Cargo at:
@temporalio/core-bridge:     https://www.rust-lang.org/tools/install
@temporalio/core-bridge: Did not copy "cdylib:temporal_sdk_node_bridge"
@temporalio/core-bridge: /Users/me/gh/sdk-node/packages/core-bridge/scripts/build.js:56
@temporalio/core-bridge:     throw new Error(`Failed to build${target ? ' for ' + target : ''}`);
@temporalio/core-bridge:     ^
@temporalio/core-bridge: Error: Failed to build
@temporalio/core-bridge:     at compile (/Users/me/gh/sdk-node/packages/core-bridge/scripts/build.js:56:11)
@temporalio/core-bridge:     at Object.<anonymous> (/Users/me/gh/sdk-node/packages/core-bridge/scripts/build.js:107:5)
@temporalio/core-bridge:     at Module._compile (internal/modules/cjs/loader.js:1063:30)
@temporalio/core-bridge:     at Object.Module._extensions..js (internal/modules/cjs/loader.js:1092:10)
@temporalio/core-bridge:     at Module.load (internal/modules/cjs/loader.js:928:32)
@temporalio/core-bridge:     at Function.Module._load (internal/modules/cjs/loader.js:769:14)
@temporalio/core-bridge:     at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:72:12)
@temporalio/core-bridge:     at internal/main/run_main_module.js:17:47
@temporalio/core-bridge: npm ERR! code ELIFECYCLE
@temporalio/core-bridge: npm ERR! errno 1
@temporalio/core-bridge: npm ERR! @temporalio/core-bridge@0.2.0 build-rust: `TEMPORAL_WORKER_FORCE_BUILD=true node ./scripts/build.js`
@temporalio/core-bridge: npm ERR! Exit status 1
@temporalio/core-bridge: npm ERR! 
@temporalio/core-bridge: npm ERR! Failed at the @temporalio/core-bridge@0.2.0 build-rust script.
@temporalio/core-bridge: npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
@temporalio/core-bridge: npm WARN Local package.json exists, but node_modules missing, did you mean to install?
@temporalio/core-bridge: npm ERR! A complete log of this run can be found in:
@temporalio/core-bridge: npm ERR!     /Users/me/.npm/_logs/2021-08-07T15_21_04_953Z-debug.log
@temporalio/core-bridge: npm ERR! code ELIFECYCLE
@temporalio/core-bridge: npm ERR! errno 1
@temporalio/core-bridge: npm ERR! @temporalio/core-bridge@0.2.0 build: `npm run build-rust`
@temporalio/core-bridge: npm ERR! Exit status 1
@temporalio/core-bridge: npm ERR! 
@temporalio/core-bridge: npm ERR! Failed at the @temporalio/core-bridge@0.2.0 build script.
@temporalio/core-bridge: npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
@temporalio/core-bridge: npm WARN Local package.json exists, but node_modules missing, did you mean to install?
@temporalio/core-bridge: npm ERR! A complete log of this run can be found in:
@temporalio/core-bridge: npm ERR!     /Users/me/.npm/_logs/2021-08-07T15_21_05_005Z-debug.log
lerna ERR! npm run build exited 1 in '@temporalio/core-bridge'
lerna WARN complete Waiting for 1 child process to exit. CTRL-C to exit immediately.
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! typescript-sdk@ build: `lerna run --stream build`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the typescript-sdk@ build script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/me/.npm/_logs/2021-08-07T15_21_05_054Z-debug.log
```

</details>

After following [building.md instructions](https://github.com/temporalio/sdk-node/blob/main/docs/building.md) (and doing [lerna clean -f](https://github.com/temporalio/sdk-node/issues/176)), I'm now able to build.

Do all Macs need to do building.md instructions in order to `npm run build`, or just a subset of them?

--- a/CONTRIBUTING.md
+++ b/CONTRIBUTING.md
@@ -10,23 +10,30 @@ You will need to submit a CLA before we can accept your contribution. You only h
 
 ### [SDK Structure](./docs/sdk-structure.md)
 
-### Environment set up
+### Environment setup
 
-- Install system dependencies including the rust toolchain by follow the instructions [here](https://docs.temporal.io/docs/node/getting-started#install-system-dependencies)
-- Clone the [sdk-node](https://github.com/temporalio/sdk-node) repo
-- Initialize the Core SDK submodule
+- Install the system dependencies listed in [Getting started > Step 0: Prerequisites](https://docs.temporal.io/docs/node/getting-started/#step-0-prerequisites)
+- Install the [Rust toolchain](https://rustup.rs/)
+- Clone the [sdk-node](https://github.com/temporalio/sdk-node) repo:
+  ```sh
+  git clone https://github.com/temporalio/sdk-node.git
+  cd sdk-node
+  ```
+- Initialize the Core SDK submodule:
   ```sh
   git submodule init
   git submodule update
   ```
-- Install the dependencies
+- Install the dependencies:
   ```sh
   npm ci
   ```
 
-> For cross compilation on MacOS follow [these](https://github.com/temporalio/sdk-node/blob/main/docs/building.md) instructions.
+You should now be able to successfully do `npm run build`. If this fails, resetting your environment may help: `npx lerna clean -- -f && npm ci`.
 
 To update your environment, run `git submodule update` to update to the latest version of the Core SDK, followed by `npm run build` to recompile.
+
+> For cross compilation on MacOS follow [these instructions](https://github.com/temporalio/sdk-node/blob/main/docs/building.md) (only required for publishing packages).
 
 ### Development Workflow
 

